### PR TITLE
Just some cleanup of expect

### PR DESCRIPTION
**Pull request by serathius**
cc @ahrtr

### CODE REVIEW

File: tests/framework/e2e/cluster.go [hunk 1]
```diff
@@ -524,10 +523,3 @@ func (epc *EtcdProcessCluster) Close() error {
 	epc.lg.Info("closed test cluster.")
 	return err
 }
-
-func (epc *EtcdProcessCluster) WithStopSignal(sig os.Signal) (ret os.Signal) {
```
Comment (ahrtr) [inline tests/framework/e2e/cluster.go:528]: If you remove this method, then it doesn't implement interface [EtcdProcess](https://github.com/etcd-io/etcd/blob/b886bbc89f31d7ac59bdfbb3d14a4263dae85a1b/tests/framework/e2e/etcd_process.go#L48) anymore.

Comment (ahrtr) [reply tests/framework/e2e/cluster.go:528]: It seems OK. You do not remove `(*EtcdServerProcess) WithStopSignal`

**Review by ahrtr — APPROVED** [review]
LGTM

File: tests/framework/e2e/cluster.go (hunk 1 above)
Comment (ahrtr) [reply tests/framework/e2e/cluster.go:528]:
```
 stderr: # go.etcd.io/etcd/tests/v3/framework/e2e
Error: stderr: framework/e2e/cluster_proxy.go:105:19: p.etcdProc.WithStopSignal undefined (type EtcdProcess has no field or method WithStopSignal)
Error: stderr: framework/e2e/cluster_proxy.go:158:16: pp.proc.StopSignal undefined (type *expect.ExpectProcess has no field or method StopSignal)
Error: stderr: framework/e2e/cluster_proxy.go:159:9: pp.proc.StopSignal undefined (type *expect.ExpectProcess has no field or method StopSignal)
```

File: tests/framework/e2e/etcd_process.go [hunk 2]
```diff
@@ -163,12 +162,6 @@ func (ep *EtcdServerProcess) Close() error {
 	return nil
 }
 
-func (ep *EtcdServerProcess) WithStopSignal(sig os.Signal) os.Signal {
```
Comment (ahrtr) [inline tests/framework/e2e/etcd_process.go:166]: You can't remove this method directly. 

```
 stderr: # go.etcd.io/etcd/tests/v3/framework/e2e
Error: stderr: framework/e2e/cluster_proxy.go:105:19: p.etcdProc.WithStopSignal undefined (type EtcdProcess has no field or method WithStopSignal)
Error: stderr: framework/e2e/cluster_proxy.go:158:16: pp.proc.StopSignal undefined (type *expect.ExpectProcess has no field or method StopSignal)
Error: stderr: framework/e2e/cluster_proxy.go:159:9: pp.proc.StopSignal undefined (type *expect.ExpectProcess has no field or method StopSignal)

```

**Review by ahrtr — CHANGES_REQUESTED** [review]

File: tests/framework/e2e/etcd_process.go (hunk 2 above)
Comment (serathius) [reply tests/framework/e2e/etcd_process.go:166]: Missed this due to `proxy` having a different build target :(

**Review by ahrtr — APPROVED** [review]
LGTM

**Pull request merged**